lib/matplotlib/cbook.py

```diff
@@ -2487,6 +2490,41 @@ def pts_to_midstep(x, *args):
                    'step-post': pts_to_poststep,
                    'step-mid': pts_to_midstep}
 
+
+def get_index_y(y):
```

Comment:
I have a commit to change this to `index_of` which I agree is better, but still not perfect. It returns both the index and the values (the original name was going for 'get (index, y)' so something that sounds like it should return both would be better (`with_index` sounds like it should be a context manager, `indexed` is too ambiguous).